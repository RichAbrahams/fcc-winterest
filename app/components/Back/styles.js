import styled from 'styled-components';
import FontAwesome from 'react-fontawesome';

export const FA = styled(FontAwesome) `
  color: white;
  text-shadow: 0px 8px rgba(21,16,16, 0.5);
  &:hover {
    color: #48e248;
  }
`;
