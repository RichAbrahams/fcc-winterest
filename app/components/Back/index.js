/**
*
* Back
*
*/

import React from 'react';
import { browserHistory } from 'react-router';
import { FA } from './styles';

function Back() {
  return (
    <FA
      name="arrow-circle-o-left"
      size="4x"
      onClick={() => browserHistory.goBack()}
    />
  );
}

Back.propTypes = {

};

export default Back;
