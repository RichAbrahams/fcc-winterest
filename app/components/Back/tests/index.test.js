import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Back from '../index';

const mockFunction = jest.fn();

it('renders correctly', () => {
  const component = renderer.create(
    <Back toggleAddWin={mockFunction} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should be able to click', () => {
  const element = mount(<Back toggleAddWin={mockFunction} />);
  element.find('span').simulate('click');
  expect(mockFunction).toHaveBeenCalled();
});
