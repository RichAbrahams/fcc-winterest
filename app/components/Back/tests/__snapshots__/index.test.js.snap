exports[`test renders correctly 1`] = `
<span
  aria-hidden={true}
  className="fa fa-arrow-circle-o-left fa-4x VmiKC"
  onClick={[Function]} />
`;
